plexapi/library.py

```diff
@@ -436,21 +445,83 @@ def reload(self, key=None):
         return self._server.library.section(self.title)
 
     def edit(self, agent=None, **kwargs):
-        """ Edit a library (Note: agent is required). See :class:`~plexapi.library.Library` for example usage.
+        """ Edit a library. See :class:`~plexapi.library.Library` for example usage.
 
             Parameters:
+                agent (str, optional): The library agent.
                 kwargs (dict): Dict of settings to edit.
         """
         if not agent:
             agent = self.agent
-        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(kwargs))
+
+        locations = []
+        if kwargs['location']:
+            for path in kwargs['location']:
+                if not self._server.isBrowsable(path):
+                    raise BadRequest('Path: %s does not exist.' % path)
+            locations = [('location', path) for path in kwargs.pop('location')]
```

Comment:
Append locations while checking browsable so you only loop once.

```suggestion
                locations.append(('location', path))
```

Reply:
We need to check for locations that are str in order to loop properly.
```suggestion
        locations = []
        if kwargs['location']:
            if isinstance(kwargs['location'], str):
                kwargs['location'] = [kwargs['location']]
            for path in kwargs.pop('location'):
                if not self._server.isBrowsable(path):
                    raise BadRequest('Path: %s does not exist.' % path)
                locations.append(('location', path))
```